syntax = "proto3";

package ray.rpc;

import "src/ray/protobuf/common.proto";

message AssignTaskRequest {
  // The task to be pushed.
  Task task = 1;
  // A list of the resources reserved for this worker.
  repeated ResourceIdSetInfo resource_ids = 2;
}

message AssignTaskReply {
}

// Service for worker.
service WorkerTaskService {
  // Push a task to a worker.
  rpc AssignTask(AssignTaskRequest) returns (AssignTaskReply);
}
